frontend/test/metabase/dashboard/dashboard.cy.spec.js

```diff
@@ -112,4 +112,59 @@ var iframeUrl = METABASE_SITE_URL + "/embed/dashboard/" + token + "#bordered=tru
     cy.contains("Remove").click();
     cy.contains("Save").click();
   });
+
+  it("should parse duplicate parameter query params", () => {
+    cy.visit("/browse/1");
+    cy.contains("Orders").click();
+    cy.contains("Save").click();
+    modal()
+      .contains("button", "Save")
+      .click();
+    modal()
+      .contains("Yes please!")
+      .click();
+    modal()
+      .contains("Create a new dashboard")
+      .click();
+    modal()
+      .contains("Name")
+      .next()
+      .type("my dash");
+    cy.url().as("questionUrl"); // this is for resetting
+    modal()
+      .contains("Create")
+      .click();
+
+    // add an id parameter
+    cy.get(".Icon-funnel_add").click();
+    popover()
+      .contains("ID")
+      .click();
+    cy.contains("Select…").click();
+    popover()
+      .contains("ID")
+      .first()
+      .click({ force: true });
+    cy.contains("Done").click();
+    cy.contains("Save").click();
```

Comment:
If we **are** going to add tests like this, I think it would be much better if this whole thing was replaced with calls to `cy.request("POST", "/api/card")` and `cy.request("POST", "/api/dashboard")`. Maybe add some helpers to `__support__/cypress`/

Basically we shouldn't be testing unrelated UI over and over again in tests like this, since if something about it changes we'll have to fix a bunch of unrelated tests.

Reply:
I was originally going to add this parameterized dashboard to a snapshot (probably default), but this was on release rather than master.

Though, your point remains if this code is in a snapshot creator. It should still be a POST rather than button clicks.